torchvision/prototype/features/_encoded.py

```diff
@@ -16,7 +16,9 @@
 class EncodedData(_Feature):
     @classmethod
     def _wrap(cls: Type[D], tensor: torch.Tensor) -> D:
-        return tensor.as_subclass(cls)
+        output = tensor.as_subclass(cls)
+        output._tensor = tensor
```

Comment:
Originally you had concerns adopting this approach in order to avoid issues with in-place operators. Assuming we keep this reference private, are we safe from in-place operators? Aka do the shapes of `img` and `img._tensor` match on the following?
```
x = torch.randn((10,1,1,1))
img = Image(x)
img.resize_((10, 1, 10, 10))
```

Reply:
No, we are not safe with `resize_`-like ops and the shapes wont match...
This is a drawback of using tensor ref...

Reply:
Then probably that's something we would like to avoid, right? Shall we proceed with your original proposal (https://github.com/pytorch/vision/pull/6681#issuecomment-1268703395) to use `DisableTorchFunction`? My understanding is that this would be safe for inplace ops with little performance penalty?

Reply:
Yes, it would be better to avoid that. I'll check DisableTorchFunction again but seeing that the gain is very tiny and around benchmarking noise maybe we can search for other optims now

Reply:
I'm open to it if we can guarantee it's safe. If we go ahead with keeping a reference, is it possible to do a bit of refactoring to use `__` and make it much harder for users to shoot themselves in the foot? I suspect the issue here is subclassing of our `_Feature` but I'm open to ideas.

@vfdev-5 WDYT of Philip's proposal?

Reply:
I check that later, but as commented previously, I do not see that this update brings much perf gain, just very tiny speed-up. Let me come back to this PR later. I'll put it as draft now.